hathor/transaction/storage/transaction_storage.py

```diff
@@ -584,7 +625,15 @@ def get_all_tips(self, timestamp: Optional[float] = None) -> Set[Interval]:
         assert self.all_index is not None
         if timestamp is None:
             timestamp = self.latest_timestamp
+
+        if self._all_tips_cache is not None and timestamp >= self._all_tips_cache.timestamp:
```

Comment:
It feels wrong to me. I think we should compare to `self.latest_timestamp`.

Reply:
Every time a new tx arrives, the cache is invalidated. So the cache will always be up to date with `self.latest_timestamp`, ie `self._all_tips_cache.timestamp == self.latest_timestamp`.

Reply:
Yes, that's the idea. `self.latest_timestamp` is supposed to be equal to `self._all_tips_cache.timestamp` if there is a `self._all_tips_cache`

Reply:
In this case, we should add an assert to double check it.

Reply:
Added the assert